internal/server/filesystem/server.go

```diff
@@ -2,24 +2,227 @@ package filesystem
 
 import (
 	"context"
+	"fmt"
+	"regexp"
+	"strings"
 
-	"github.com/kubernetes-csi/csi-proxy/client/api"
 	"github.com/kubernetes-csi/csi-proxy/client/apiversion"
 	"github.com/kubernetes-csi/csi-proxy/internal/server/filesystem/internal"
 )
 
-// Server is the internal server for the filesytem API group.
-type Server struct{}
+type Server struct {
+	kubeletCSIPluginsPath string
+	kubeletPodPath        string
+	hostAPI               API
+}
+
+const MAX_PATH_LENGTH_WINDOWS = 260
+
+var invalidPathCharsRegexWindows = regexp.MustCompile(`["/\:\?\*|]`)
+
+type API interface {
+	PathExists(path string) (bool, error)
+	Mkdir(path string) error
+	Rmdir(path string) error
+	LinkPath(tgt string, src string) error
+}
+
+func NewServer(hostOS string, kubeletCSIPluginsPath string, kubeletPodPath string, hostAPI API) (*Server, error) {
+	if hostOS != "windows" {
```

Comment:
I'm sorry, I'm not sure I get the idea behind this; could you please elaborate a mind what you have in mind? It seems it never gets called with anything other than windows, even for tests, which I thought was the idea in the 1st place?

Reply:
I agree this is useless. It's an artifact of something else I was trying and will remove.